back/engines/commercial/analysis/app/lib/analysis/auto_tagging_method/base.rb

```diff
@@ -160,6 +160,21 @@ def deduct_locale(input)
         AppConfiguration.instance.settings('core', 'locales').first
     end
 
+    def classification_fields
+      fields = analysis.associated_custom_fields
+      if fields.map(&:code).include?('topic_ids') || analysis.participation_method != 'ideation'
+        fields
+      else
+        custom_form = analysis.project.custom_form || CustomForm.new(participation_context: analysis.project)
```

Comment:
What is the difference between:
- `analysis.project.custom_form` and
- `CustomForm.new(participation_context: analysis.project)`?

Are both necessary?

Reply:
Sadly, we have this code in many places. When a persisted custom form does not exist, we need to fall back to a default form. Ideally, we will start looking into the broader topic of custom fields and tech debt during this tandem https://www.notion.so/citizenlab/Quality-Quest-24Q2-443732b60cb74d8eb455fc12ab57e330?pvs=4#641a64cbe3074efd8b61008972ab4fc7 and I wrote this down, as this clearly indicates that we need at least a helper method for this.